torch/csrc/autograd/engine.cpp

```diff
@@ -375,6 +375,10 @@ auto Engine::thread_main(const std::shared_ptr<GraphTask>& graph_task) -> void {
           // queue_callback() to find the target GraphTask to append final
           // callbacks.
           GraphTaskGuard guard(local_graph_task);
+          NodeGuard ndguard;
```

Comment:
Looking at this, it feel like adding the extra method and check might actually be more expensive that just doing the assignment of the thread local.
I think you can just pass the Node to the constructor and always set the thread local properly, ignoring the current anomaly mode state.

Reply:
Yep, I agree with you. Will change it in the next commit